File: infrastructure/spring-boot-webflux-rest-api-adapter/src/main/java/codes/fdk/blueprint/api/infrastructure/spring/rest/webflux/PostCategoryRequest.java
package codes.fdk.blueprint.api.infrastructure.spring.rest.webflux;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Pattern;

record PostCategoryRequest(@JsonProperty("name") @NotBlank String name,
                           @JsonProperty("slug") @Pattern(regexp = "[^\s]+") @NotBlank String slug,
                           @JsonProperty("visible") boolean visible) {}

File: infrastructure/vertx-web-openapi-adapter/src/main/kotlin/codes/fdk/blueprint/api/infrastructure/web/openapi/PostCategoryRequest.kt
package codes.fdk.blueprint.api.infrastructure.web.openapi

import com.fasterxml.jackson.annotation.JsonProperty

data class PostCategoryRequest(
    @JsonProperty("name") val name: String,
    @JsonProperty("slug") val slug: String,
    @JsonProperty("visible") val visible: Boolean
)
